MachOKit/Core/Impl/MKPtr.m

```diff
@@ -66,6 +67,20 @@
 {
     NSCParameterAssert([node isKindOfClass:MKBackedNode.class]);
     
+    // If the image has been processed by dyld, the pointer points to the actual
+    // (slid) address of the pointee. MKPtr instead expects the address to follow
+    // MKNodeVMAddress semantics, so we must adjust for that.
+    // TODO - Add a test for this situation, and check if there are edge cases
+    // where dyld doesn't slide the pointer
+    MKMachOImage *image = [node nearestAncestorOfType:MKMachOImage.class];
+    if (image && image.isFromMemory) {
+        // TODO - For some reason, mk_vm_address_remove_slide thinks that
+        // `(mk_vm_address_t)(slide * -1) > addr` is true, so it returns
+        // MK_EUNDERFLOW. I'm not sure whether that's a bug here or in
+        // that function, but it should probably be looked into.
```

Comment:
It is a bug in `mk_vm_address_remove_slide`.  Thanks!

Reply:
Would it make sense to use `__builtin_*_overflow` for overflow-related arithmetic operations? Those might be more efficient, assuming Clang doesn't realise that you're performing an overflow check the manual way

Reply:
Yes.  Mach-O Kit should switch to the clang built-ins.